src/extension.js

```diff
@@ -0,0 +1,52 @@
+const vscode = require('vscode');
+const child_process = require('child_process');
+const util = require('util');
+
+const exec = util.promisify(child_process.exec);
+
+const extensionName = "hledger";
+
+function activate(context) {
+  const accountAutocompleteProviderSubscription =
+    vscode.languages.registerCompletionItemProvider("hledger", {
+      async provideCompletionItems(
+        document,
+        position,
+        token,
+        context
+      ) {
+        const text = document.lineAt(position).text;
+        if (!text.match(/^\s/)) {
+          // Do not provide account name completions if the line is not indented.
+          return;
+        }
+
+        const configuration = vscode.workspace.getConfiguration();
+        const hledgerPath = configuration.get(
+          `${extensionName}.hledgerPath`,
+          "hledger"
+        );
+        const ledgerFile = configuration.get(`${extensionName}.ledgerFile`, "");
+
+        const env = process.env;
+        if (ledgerFile) {
+          env.LEDGER_FILE = ledgerFile;
+        }
+        const { stdout } = await exec(`"${hledgerPath}" accounts`, { env });
```

Comment:
For me, running this takes:

```
$ time hledger accounts
586 millis
```

And it took a bit longer before I upgraded to the M1. I just wonder if we should perhaps cache this? Or have some strategy for if it takes a while? What do you think?